docs/rfcs/rfc-025-cypher-field-filters.md

```diff
@@ -0,0 +1,302 @@
+            name
+            year
+        }
+    }
+}
+```
+
+**Current mostRecentBooks predicate**
+
+```cypher
+single(this0 IN [(this)-[:AUTHORED_BOOK]->(this0:`Book`) | this0] WHERE this0.name = "Simone's adventures!")
+```
+
+As the `mostRecentBooks` is already resolved it's possible to filter it directly without the pattern matching, as follows:
+
+**Cypher Proposed**
+
+```cypher
+MATCH (this:`Author`)
+CALL {
+    WITH this
+    CALL {
+        WITH this
+        MATCH (this)-[:AUTHORED_BOOK]->(b:Book) RETURN b AS result ORDER BY b.year DESC LIMIT 1
+    }
+    WITH result AS this_mostRecentBooks
+    RETURN head(collect(this_mostRecentBooks { .name, .year })) AS this_mostRecentBooks
+}
+WITH *
+WHERE (this.name = "Simone" AND single(var in this_mostRecentBooks WHERE var.name = "Simone's adventures!"))
+RETURN this { .name, mostRecentBooks: this_mostRecentBook } AS this
+```
+
+#### ALL filter
+
+The `_ALL` filter is based on a combination of the `EXISTS` sub-clause, this behavior has to be changed for `@cypher` resolved fields.
+
+**GraphQL**
+
+```graphql
+{
+    authors(where: { name: "Simone", mostRecentBooks_ALL: { name: "Simone's adventures!" } }) {
+        name
+        mostRecentBooks {
+            name
+            year
+        }
+    }
+}
+```
+
+**Current mostRecentBooks predicate**
+
+```cypher
+WHERE (EXISTS {
+    MATCH (this)-[:AUTHORED_BOOK]->(this0:`Book`)
+    WHERE this0.name = "Simone's adventures!"
+} AND NOT (EXISTS {
+    MATCH (this)-[:AUTHORED_BOOK]->(this0:`Book`)
+    WHERE NOT (this0.name = "Simone's adventures!"
+}))
+```
+
+As the `mostRecentBooks` is already resolved it's possible to filter it directly without the `EXISTS` sub-clause,
+but by using the `all` predicate function as follow:
+
+**Cypher Proposed**
+
+```cypher
+MATCH (this:`Author`)
+CALL {
+    WITH this
+    CALL {
+        WITH this
+        MATCH (this)-[:AUTHORED_BOOK]->(b:Book) RETURN b AS result ORDER BY b.year DESC LIMIT 1
+    }
+    WITH result AS this_mostRecentBooks
+    RETURN head(collect(this_mostRecentBooks { .name, .year })) AS this_mostRecentBooks
+}
+WITH *
+WHERE (this.name = "Simone" AND all(var in this_mostRecentBooks WHERE var.name = "Simone's adventures!"))
+RETURN this { .name, mostRecentBooks: this_mostRecentBook } AS this
+```
+
+## Challenges
+
+To apply filters to the `@cypher` fields, it will be required to have already these resolved.
+It will be efficient to resolve these fields only when are required for some filters or/and are present in the `SelectionSet` rather than resolve these all the time.
+Also changing the translation order of these fields may be not trivial.
+
+## Risks
+
+-   This solution is relying on the new less used and tested `columnName` feature.
+-   It requires changing the translation order of the `@cypher` fields (no longer at Projection time).
+-   Maintaining some different predicates between `@cypher` field filters and standard filters.
+
+## Security consideration
+
+-   None.
```

Comment:
Although the risk is low, as `@cypher` is defined in the server, we need to keep an eye on cypher injection when making these changes